transport/src/main/java/io/netty/channel/nio/SelectedSelectionKeySet.java

```diff
@@ -107,4 +62,25 @@ public boolean contains(Object o) {
     public Iterator<SelectionKey> iterator() {
         throw new UnsupportedOperationException();
     }
+
+    void reset() {
+        reset(0);
+    }
+
+    void reset(int start) {
+        for (; start < size; ++start) {
+            keys[start] = null;
+        }
+        size = 0;
```

Comment:
I think so. As we iterate over the array in `NioEventLoop` we null entries out. This method is used if we decide we are done iterating over the array and want to make the set ready for re-use. Do you see an issue here?